samza-core/src/main/java/org/apache/samza/coordinator/leaderelection/LeaderElector.java

```diff
@@ -20,6 +20,8 @@
 package org.apache.samza.coordinator.leaderelection;
 
 import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.coordinator.LeaderElectorListener;
+
 
 @InterfaceStability.Evolving
 public interface LeaderElector {
```

Comment:
I need an api to cancel the leaderElector. AppRunner only needs one time leader election, so the code will look like:

leaderElector.tryBecomeLeader(new Listner() {
  void onBecomeLeader()  {
    do stream creation;
    release latch;
    leaderElector.cancel();
  }
);